### PR TITLE
Integrate Example from README into Tests

### LINKED ISSUE #51: Integrate Example from README into Tests
To ensure the documentation remains up-to-date and accurate, we need to incorporate the example provided in the README file into our test suite. This will allow us to verify that the documented example is always functioning as expected and will alert us to the need for documentation updates if changes occur.

### PR DESCRIPTION
**Pull request by ksew1**
Closes #51

### CODE REVIEW

File: crates/cairo-coverage/tests/data/readme_example/.snfoundry_versioned_programs/readme_example_Tests.sierra.json
Comment (ddoktorski) [inline crates/cairo-coverage/tests/data/readme_example/.snfoundry_versioned_programs/readme_example_Tests.sierra.json:1]: Wouldn't it be better to add logic that compiles this example and saves trace data instead of committing these files?

File: crates/cairo-coverage/tests/data/readme_example/Scarb.lock
Comment (ddoktorski) [inline crates/cairo-coverage/tests/data/readme_example/Scarb.lock:1]: nit: No need to commit `Scarb.lock` file

File: crates/cairo-coverage/tests/data/readme_example/.snfoundry_versioned_programs/readme_example_Lib.sierra.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    "annotations": {
+      "github.com/software-mansion/cairo-profiler": {
+        "statements_functions": {}
+      },
+      "github.com/software-mansion/cairo-coverage": {
+        "statements_code_locations": {}
+      }
+    }
+  }
+}
```
Comment (ddoktorski) [inline crates/cairo-coverage/tests/data/readme_example/.snfoundry_versioned_programs/readme_example_Lib.sierra.json:20]: nit: Newline 👀

File: crates/cairo-coverage/tests/helpers/test_project.rs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        SnapboxCommand::new(cargo_bin!("cairo-coverage"))
+            .args(&trace_files)
+            .args(args)
+            .current_dir(&self.dir)
+            .assert()
+            .success();
+        Ok(self)
+    }
+
+    fn run_genhtml(self) -> Self {
```
Comment (ddoktorski) [inline crates/cairo-coverage/tests/helpers/test_project.rs:61]: Is it necessary?

File: crates/cairo-coverage/tests/output/complex_calculator.lcov
Comment (ddoktorski) [inline crates/cairo-coverage/tests/output/complex_calculator.lcov:1]: nit: What about renaming directory `output` -> `expected_output`?

File: crates/cairo-coverage/src/input/sierra_to_cairo_map.rs [hunk 3]
```diff
@@ -61,7 +61,10 @@ fn find_statement_origin(
         .zip(function_names)
         // TODO: We should probably filter by path to user project not by path to cache
         // TODO: Can get this from source_sierra_path in call trace
-        .find(|((file_location, _), _)| !file_location.contains("com.swmansion.scarb"))
+        .find(|((file_location, _), _)| {
+            !file_location.contains("com.swmansion.scarb")
+                && !file_location.contains(".cache/scarb")
```
Comment (ddoktorski) [inline crates/cairo-coverage/src/input/sierra_to_cairo_map.rs:66]: I guess this is not strictly related to this PR?

File: crates/cairo-coverage/tests/data/readme_example/Scarb.toml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+[dev-dependencies]
+snforge_std = { git = "https://github.com/foundry-rs/starknet-foundry", tag = "v0.29.0" }
+
+[[target.starknet-contract]]
+sierra = true
+
+[scripts]
+test = "snforge test"
+
+[cairo]
```
Comment (ddoktorski) [inline crates/cairo-coverage/tests/data/readme_example/Scarb.toml:19]: nit: To align with the recommended configuration
```suggestion
[profile.dev.cairo]
```

File: crates/cairo-coverage/tests/helpers/test_project.rs [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+            .context("Failed to read the directory for trace files")?
+            .map(|entry| {
+                entry
+                    .context("Failed to read a directory entry")
+                    .map(|e| e.path().display().to_string())
+            })
+            .collect()
+    }
+
+    fn output_lcov_path(&self) -> PathBuf {
```
Comment (ddoktorski) [inline crates/cairo-coverage/tests/helpers/test_project.rs:46]: With the current implementation, it seems quite tricky to test anything using the `--output-path` flag. Maybe it would be better to modify it to be more flexible?

File: crates/cairo-coverage/tests/e2e/general.rs [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+#[test]
+fn readme_example() {
+    // If you ever find yourself in a situation where you need to change the expected output,
+    // please update the lcov.md files as well.
+    TestProject::run("readme_example").output_same_as_in_file("readme_example.lcov");
+}
+
+#[test]
+fn no_trace_files_provided() {
```
Comment (ddoktorski) [inline crates/cairo-coverage/tests/e2e/general.rs:36]: I would suggest removing this test because it actually tests the clap functionality, but leave this up to you.

File: crates/cairo-coverage/src/input/sierra_to_cairo_map.rs (hunk 3 above)
Comment (ksew1) [reply crates/cairo-coverage/src/input/sierra_to_cairo_map.rs:67]: This fix ensures that the tests pass on the CI. It will be redesigned in issue [#55](https://github.com/software-mansion/cairo-coverage/issues/55).

File: crates/cairo-coverage/tests/helpers/test_project.rs (hunk 5 above)
Comment (ksew1) [reply crates/cairo-coverage/tests/helpers/test_project.rs:62]: Change that and added test for this flag

File: crates/cairo-coverage/tests/helpers/test_project.rs (hunk 2 above)
Comment (ksew1) [reply crates/cairo-coverage/tests/helpers/test_project.rs:85]: This helps because what we output may not have any semantic sense, and `genhtml` verifies that. However, `genhtml` completely depends on the contents of the lcov file. So if the output of the tests doesn't change, the `genhtml` output also won't change. This means that you only need to test the output once when you change it. But then the developer needs to remember to test it with `genhtml` after any changes. Rather than relying on the developer, I would prefer to run this every time to enforce it.

Comment (ddoktorski) [reply crates/cairo-coverage/tests/helpers/test_project.rs:85]: It would be nice to add some kind of development guideline to have all the required tools for development listed in one place

**Review by ddoktorski — APPROVED** [review]
LGTM 💪

File: crates/cairo-coverage/tests/helpers/test_project.rs (hunk 2 above)
Comment (ksew1) [reply crates/cairo-coverage/tests/helpers/test_project.rs:85]: Will add contribution guide
